utils/db/teams.js

```diff
@@ -135,6 +148,27 @@ export async function getTeamsForUserId(userId) {
       team.users = augmentedUsersArr;
     })
   );
-  // console.log(teams)
   return teams;
 }
+
+// callback runs whenever teams changes
+// callback is passed an array of teams
+export async function getTeamsForUserId(userId, callback) {
+  let teams = [];
+
+  // Call callback whenever teams change
+  const teamIdsSnapshot = firebase.database().ref(`/users/${userId}/teams`);
+
+  teamIdsSnapshot.on("value", async snapshot => {
+    let teamIds = snapshot.val();
+    if (!teamIds) {
+      callback([]);
+      return;
+    }
+    // Workaround for weird issue that had some teamIds set as undefined
+    teamIds = teamIds.filter(teamId => teamId != undefined);
+    teams = await getTeams(teamIds);
+    teams = await augmentTeamsWithUserNames(teams);
```

Comment:
Minor, would prefer the following for readability:
```
teams = await augmentTeamsWithUserNames(
        await getTeams(teamIds)
);
```

idk if this is good async await notation though